app/controllers/miq_policy_controller/alert_profiles.rb

```diff
@@ -343,4 +347,28 @@ def alert_profile_get_info(alert_profile)
     @right_cell_text = _("%{model} \"%{name}\"") % {:model => ui_lookup(:model => "MiqAlertSet"), :name => alert_profile.description}
     @right_cell_div = "alert_profile_details"
   end
+
+  def alert_profile_sync_provider(old_alerts = nil, new_alerts = nil)
+    if @alert_profile.mode == "MiddlewareServer"
+      if old_alerts.nil? && new_alerts.nil?
+        operation = :update_assignments
+        old_alerts = new_alerts = @alert_profile.miq_alerts.collect(&:id)
+        assigned = @alert_profile.get_assigned_tos
+      else
+        operation = :update_alerts
+      end
+      MiqQueue.put(
+        :class_name  => "ManageIQ::Providers::Hawkular::MiddlewareManager",
```

Comment:
can this be `ManageIQ::Providers::MiddlewareManager` ? To make it provider agnostic?